internal/adaptorsigs/adaptor_test.go

```diff
@@ -0,0 +1,131 @@
+package adaptorsigs
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4/schnorr"
+)
+
+func TestAdaptorSignatureRandom(t *testing.T) {
+	seed := time.Now().Unix()
+	rng := rand.New(rand.NewSource(seed))
+	defer func(t *testing.T, seed int64) {
+		if t.Failed() {
+			t.Logf("random seed: %d", seed)
+		}
+	}(t, seed)
+
+	for i := 0; i < 100; i++ {
+		// Generate two private keys
```

Comment:
It doesn't matter much since this is test code, but you could simplify this quite a bit with:

```Go
		// Generate two private keys.
		privKey1, err := secp256k1.GeneratePrivateKeyFromRand(rng)
		if err != nil {
			t.Fatalf("failed to read random private key: %v", err)
		}
		privKey2, err := secp256k1.GeneratePrivateKeyFromRand(rng)
		if err != nil {
			t.Fatalf("failed to read random private key: %v", err)
		}
```

That would also technically fix a bug with your test code too.  Namely, 0 is not a valid private key.  Of course the chances of actually randomly generating 0 here are essentially impossible, but still.